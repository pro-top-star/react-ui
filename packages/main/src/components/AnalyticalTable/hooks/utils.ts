export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
