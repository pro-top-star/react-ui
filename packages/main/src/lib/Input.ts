import { Input } from '../webComponents/Input';

export { Input };
