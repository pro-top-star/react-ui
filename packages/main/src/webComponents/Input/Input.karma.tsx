import { mountThemedComponent } from '@shared/tests/utils';
import { expect, use } from 'chai';
import { matchSnapshot } from 'chai-karma-snapshot';
import React from 'react';
import { Input } from '../../lib/Input';

use(matchSnapshot);

describe('Input', () => {
  it('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<Input />);
    expect(wrapper.debug()).to.matchSnapshot();
  });
});
