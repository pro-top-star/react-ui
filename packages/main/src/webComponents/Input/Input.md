[Link to UI5 Web Components Playground](https://sap.github.io/ui5-webcomponents/playground/components/Input)

```jsx
import { Input } from '@ui5/webcomponents-react/lib/Input';
```
