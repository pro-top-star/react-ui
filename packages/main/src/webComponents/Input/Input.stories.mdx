import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs/blocks';
import { Input } from '@ui5/webcomponents-react/lib/Input';
import { InputType } from '@ui5/webcomponents-react/lib/InputType';
import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import '@ui5/webcomponents-icons/dist/icons/employee.js';
import { Icon } from '@ui5/webcomponents-react/lib/Icon';
import { CSSProperties } from 'react';
import { SuggestionItem } from '@ui5/webcomponents-react/lib/SuggestionItem';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { DocsHeader } from '@shared/stories/DocsHeader';
import { DocsCommonProps } from '@shared/stories/DocsCommonProps';

<Meta
  title="UI5 Web Components / Input"
  component={Input}
  subcomponents={{ SuggestionItem }}
  argTypes={{
    ...createSelectArgTypes({ type: InputType, valueState: ValueState }),
    children: { control: { disable: true } },
    icon: { control: { disable: true } },
    valueStateMessage: { control: { disable: true } },
    ...DocsCommonProps
  }}
  args={{
    type: InputType.Text,
    valueState: ValueState.None,
    icon: <Icon name="employee" />,
    style: {},
    className: '',
    tooltip: '',
    slot: '',
    ref: null
  }}
/>

<DocsHeader />

<Canvas>
  <Story name="Default">
    {(args) => {
      return <Input {...args} />;
    }}
  </Story>
</Canvas>

<ArgsTable story="." />
