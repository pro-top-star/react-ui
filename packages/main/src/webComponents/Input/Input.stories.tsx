import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { Input } from '@ui5/webcomponents-react/lib/Input';
import { InputType } from '@ui5/webcomponents-react/lib/InputType';
import { SuggestionItem } from '@ui5/webcomponents-react/lib/SuggestionItem';
import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import React from 'react';

export default {
  title: 'UI5 Web Components / Input',
  component: Input,
  parameters: {
    subcomponents: { SuggestionItem }
  },
  argTypes: {
    ...createSelectArgTypes({ type: InputType, valueState: ValueState }),
    children: {
      type: null
    },
    ref: {
      type: null
    },
    icon: {
      type: null
    }
  },
  args: {
    placeholder: 'Placeholder...',
    type: InputType.Text,
    valueState: ValueState.None
  }
};

export const generatedDefaultStory = (props) => (
  <Input
    disabled={props.disabled}
    maxlength={props.maxlength}
    name={props.name}
    placeholder={props.placeholder}
    readonly={props.readonly}
    required={props.required}
    showSuggestions={props.showSuggestions}
    type={props.type}
    value={props.value}
    valueState={props.valueState}
    onChange={props.onChange}
    onInput={props.onInput}
    onSubmit={props.onSubmit}
    onSuggestionItemSelect={props.onSuggestionItemSelect}
    icon={null}
    valueStateMessage={null}
  >
    Some Content
  </Input>
);

generatedDefaultStory.storyName = 'Generated default story';
