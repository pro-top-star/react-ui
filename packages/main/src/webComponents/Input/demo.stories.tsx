import { action } from '@storybook/addon-actions';
import { boolean, select } from '@storybook/addon-knobs';
import { Input } from '@ui5/webcomponents-react/lib/Input';
import { InputType } from '@ui5/webcomponents-react/lib/InputType';
import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import React from 'react';
import notes from './Input.md';

export default {
  title: 'UI5 Web Components / Input',
  component: Input,
  parameters: { notes }
};

export const generatedDefaultStory = () => (
  <Input
    disabled={boolean('disabled', false)}
    placeholder={'generatedString'}
    readonly={boolean('readonly', false)}
    type={select('type', InputType, null)}
    value={'generatedString'}
    valueState={select('valueState', ValueState, null)}
    showSuggestions={boolean('showSuggestions', false)}
    onChange={action('onChange')}
    onInput={action('onInput')}
    onSubmit={action('onSubmit')}
    onSuggestionItemSelect={action('onSuggestionItemSelect')}
    icon={null}
  >
    Input
  </Input>
);

generatedDefaultStory.story = {
  name: 'Generated default story'
};
