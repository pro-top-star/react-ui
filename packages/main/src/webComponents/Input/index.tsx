import { Event } from '@ui5/webcomponents-react-base/lib/Event';
import { InputType } from '@ui5/webcomponents-react/lib/InputType';
import { ValueState } from '@ui5/webcomponents-react/lib/ValueState';
import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import UI5Input from '@ui5/webcomponents/dist/Input';
import React, { FC, ReactNode } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface InputPropTypes extends WithWebComponentPropTypes {
  disabled?: boolean; // @generated
  placeholder?: string; // @generated
  readonly?: boolean; // @generated
  type?: InputType; // @generated
  value?: string; // @generated
  valueState?: ValueState; // @generated
  name?: string; // @generated
  showSuggestions?: boolean; // @generated
  onChange?: (event: Event) => void; // @generated
  onInput?: (event: Event) => void; // @generated
  onSubmit?: (event: Event) => void; // @generated
  onSuggestionItemSelect?: (event: Event) => void; // @generated
  icon?: ReactNode; // @generated
  children?: string;
}

/**
 * <code>import { Input } from '@ui5/webcomponents-react/lib/Input';</code>
 * <br />
 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/Input" target="_blank">UI5 Web Components Playground</a>
 */
const Input: FC<InputPropTypes> = withWebComponent<InputPropTypes>(UI5Input);

Input.displayName = 'Input';

Input.defaultProps = {
  placeholder: null, // @generated
  type: InputType.Text, // @generated
  value: '', // @generated
  valueState: ValueState.None // @generated
};

export { Input };
