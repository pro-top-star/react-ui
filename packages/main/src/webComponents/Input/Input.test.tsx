import { mountThemedComponent } from '@shared/tests/utils';
import { Input } from '@ui5/webcomponents-react/lib/Input';
import React from 'react';

describe('Input', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<Input />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
