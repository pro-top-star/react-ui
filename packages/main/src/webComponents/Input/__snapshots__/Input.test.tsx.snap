// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Basic Test (generated) 1`] = `
<ui5-input
  type="Text"
  value=""
  value-state="None"
/>
`;
